src/components/Composer/index.js

```diff
@@ -225,7 +225,13 @@ class Composer extends React.Component {
             const pastedHTML = event.clipboardData.getData(TEXT_HTML);
 
             const domparser = new DOMParser();
-            const embeddedImages = domparser.parseFromString(pastedHTML, TEXT_HTML).images;
+            const parsedHTML = domparser.parseFromString(pastedHTML, TEXT_HTML);
+            if (parsedHTML.body.childNodes.length === 1 && parsedHTML.body.childNodes[0].tagName.toLowerCase() === 'img' && files.length > 0) {
+                this.props.onPasteFile(event.clipboardData.files[0]);
+                return;
+            }
```

Comment:
This condition is really long and complicated, can we add a comment above it explaining exactly what we're doing here?

Reply:
Added.